src/iam.ts

```diff
@@ -301,6 +289,25 @@ export class IAM extends IAMBase {
     throw new Error(ERROR_MESSAGES.CLAIMS_NOT_INITIALIZED);
   }
 
+  private async getClaimId({ claimData }: { claimData: ClaimData }) {
+    const { service = [] } = await this.getDidDocument();
+    const { id, claimTypeVersion } =
+      service.find(
+        ({ profile, claimType, claimTypeVersion }) =>
+          Boolean(profile) ||
+          (claimType === claimData.claimType && claimTypeVersion === claimData.claimTypeVersion)
+      ) || {};
+
+    if (claimData.profile && id) {
+      return id;
+    }
+
+    if (claimData.claimType && id && claimData.claimTypeVersion === claimTypeVersion) {
+      return id || uuid();
+    }
+    return uuid();
```

Comment:
Picking up on Dmitry's point about two claims having the same id, I wonder if we should start doing something like hashing the `claimData` to get the id, such that we would have deterministic ids. Just a thought, maybe not necessary to do right now, but maybe now is the time to make a change like this 🤔

Reply:
👍  good idea

Reply:
@jrhender but on the other hand we have property called `hash` on the claim which represent sha256 hash of the generated token

Reply:
I think that's okay, in fact that's good, the purpose of using the hash for the `id` isn't because we don't have the hash elsewhere, it's more get deterministic ids. In this way, someone who has the generated token can check if it is revoked but hashing it and checking a revocation registry.